### PR TITLE
Milestone5

### PR DESCRIPTION
**Pull request by eypsrcnuygr**
- You can do basic math operations.
- [Here](https://my-calculator-sercan.herokuapp.com/) is the live link.

### CODE REVIEW

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @eypsrcnuygr :wave: 

Your project looks good, however, it needs some refactoring.

- [ ] First,  your **app component**. In the [project scope](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376019/), it is required that you create only a handleClick function, that accepts buttonName as an argument, your function does not accept button name as an argument.

- [ ] It is required you pass handleClick as a prop to buttonPanel not handleChange.
[screen shot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passHandleClick.png).

Kindly follow the [structure](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376019/) provided in the project requirements. This would make your code base neat and easy to follow.

- [ ]  After clearing the display component, when **AC** is left, if I perform an operation I get this error.

errrr

### That is it, implement these changes, then submit another form for a code review.

### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

File: src/Components/App.js [hunk 1]
```diff
@@ -9,78 +9,41 @@ class App extends React.Component {
...
-      total: calculatedObj.total,
-      next: calculatedObj.next,
-      operation: calculatedObj.operation,
-    }));
+    this.setState({
+      total: result.total,
+      next: result.next,
+      operation: result.operation,
+    });
   }
```
Comment (shloch) [inline src/Components/App.js]: great job

File: src/Components/ButtonPanel.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const ButtonPanel = props => {
           <Button
             name={element}
             key={firstGroup.indexOf(element)}
-            handleClick={props.ClickHandler}
+            handleClick={element => props.clickHandler(element)}
           />
```
Comment (shloch) [inline src/Components/ButtonPanel.js]: good

**Review by shloch — APPROVED** [review]
Hello @eypsrcnuygr 

Great job with the progress. I noticed you improved from the previous review.
Now we're almost done, I will disturb you further with a minor detail as we're rounding up.
Could you please handle an edge case scenario here ?. It's about 'DIVISION BY ZERO' error.
Handle that by displaying something like "`infinity`" or "`NAN`" (something like that)

[see screenshot below]

calc

**Comment by shloch** [comment]
Sorry i approved on github above by error by error, this is not approved on the dashboard, waiting for a final review with the bug mentioned above though before you get a final thumbs up

Happy coding

**Review by christianotieno — APPROVED** [review]
### Milestone Approved 🍾 
Hello @eypsrcnuygr  👋 
 ##
This milestone has met all the required changes needed for approval. 👏 
Congratulations on the hard work done on this milestone. Good job 👍 
I will go ahead and mark this as complete. Good luck with the next tasks 🥇 
##
### Happy coding 💻 
#### Christian Otieno (TSE)

**Pull request merged**